### PR TITLE
Serialize/deserialize `Block<T>` better with `System.Text.Json.JsonSerializer`

### PR DESCRIPTION
**Pull request by dahlia**
Implemented `JsonConverter` for `Block<T>` and its fields.  Here's a sample block encoded into JSON:

```json
{
  "protocolVersion": 3,
  "hash": "cf19d54cf63979d43ed9600648cf804e0971b3ec5526c80dbbf2f5e94e89f21d",
  "signature": "3045022100a69de8d30ed85f8ca62b7371baccea30ab7aeb32b8e185a8262a5ac9a05b86cf02203292da7ef37f745245e43ffe8742ab4fbf8b6fe44071b4715173435b6fc559be",
  "preEvaluationHash": "e2edbaf65d8dfd86e40fcb8f5f21bcd54333686d9262645c4a3d873021dc4f17",
  "stateRootHash": "1b16b1df538ba12dc3f97edbb85caa7050d46c148134290feba80f8236c83db9",
  "index": 0,
  "difficulty": 0,
  "totalDifficulty": 0,
  "nonce": "8a8ae289a78cc167fcfb",
  "miner": "fa7Ef0f81CBbE231aC597b3D3625D3E92C6E9611",
  "publicKey": "0365e861c270ae24ca6217c1188e405081738a93488974da9f57c00e3f0d85567e",
  "previousHash": null,
  "timestamp": "2023-01-27T10:55:14.595683+00:00",
  "txHash": "a93128c5a2ad323c81f6d6a845bb012373ceb1992d3a62477b703809e661da47",
  "transactions": [
    {
      "id": "1d6055adbcc7e23cb8b11cda11b89fede22b647d423412e2b6508f5d90fb8ac4",
      "nonce": 0,
      "signer": "fa7Ef0f81CBbE231aC597b3D3625D3E92C6E9611",
      "updatedAddresses": [],
      "signature": "MEUCIQCk8hTFAuVi1QSe+/NHfGheMXBhLIvUGNxEQmkD7kC0zgIgHNQh0zhVBbKSLQuK6oit8ghRRERoxE7nIA26C96aVwE=",
      "customActions": [],
      "timestamp": "2023-01-27T10:55:14.560103+00:00",
      "publicKey": "0365e861c270ae24ca6217c1188e405081738a93488974da9f57c00e3f0d85567e",
      "genesisHash": null
    }
  ]
}
```

### CODE REVIEW

File: Libplanet/ByteUtil.cs [hunk 1]
```diff
@@ -57,6 +60,32 @@ public static byte[] ParseHex(string hex)
             return bytes;
         }
 
+        /// <summary>
+        /// Converts a hexadecimal string to a immutable <see cref="byte"/> array.
```
Comment (moreal) [inline Libplanet/ByteUtil.cs:64]:
```suggestion
        /// Converts a hexadecimal string to an immutable <see cref="byte"/> array.
```

**Comment by dahlia** [comment]
> <https://github.com/planetarium/libplanet/actions/runs/4023856396/jobs/6915215674> Could you check this CI?

@riemannulus I addressed the issue.  Could you take a look again?

File: CHANGES.md [hunk 2]
```diff
@@ -29,17 +35,35 @@ To be released.
 
 ### Added APIs
 
+ -  Many more types became serialized and deserialized better with
+    [`System.Text.Json.JsonSerializer`] as they now have their own
+    [custom converters].  Note that these serializations are unavailable
+    on Unity due to its incomplete reflection support.  [[2756]]
```
Comment (moreal) [inline CHANGES.md:41]:
```suggestion
    on Unity due to its incomplete reflection support.  [[#2756]]
```

**Review by moreal — DISMISSED** [review]
Just typos.

**Review by riemannulus — DISMISSED** [review]

**Comment by dahlia** [comment]
@moreal All addressed.  Please take a look one more tine.  */cc @riemannulus*

**Review by moreal — APPROVED** [review]

**Review by riemannulus — APPROVED** [review]
LGTM

**Pull request merged**